### PR TITLE
[GLUTEN-7418][VL] Add checks for allocation failures and initialize variables

### PR DESCRIPTION
**Pull request by majetideepak**
## What changes were proposed in this pull request?

Add checks for allocation failures and initialize variables
Fixes: 7418

## How was this patch tested?
Build locally and on CI.

### CODE REVIEW

File: cpp/velox/udf/UdfLoader.cc [hunk 1]
```diff
@@ -96,11 +99,17 @@ std::unordered_set<std::shared_ptr<UdfLoader::UdfSignature>> UdfLoader::getRegis
 
     // Handle UDAFs.
     void* getNumUdafSym = loadSymFromLibrary(handle, libPath, GLUTEN_TOSTRING(GLUTEN_GET_NUM_UDAF), false);
+    if (getNumUdafSym == nullptr) {
+      throw gluten::GlutenException("malloc failed");
+    }
```
Comment (zhztheplayer) [inline cpp/velox/udf/UdfLoader.cc:104]: Does this change cause [this CI failure](https://github.com/apache/incubator-gluten/actions/runs/11220332046/job/31188265190?pr=7419)? 

```
org.apache.gluten.expression.VeloxUdfSuiteLocal *** ABORTED ***
  org.apache.gluten.exception.GlutenException: malloc failed
  at org.apache.gluten.udf.UdfJniWrapper.registerFunctionSignatures(Native Method)
  at org.apache.gluten.backendsapi.velox.VeloxListenerApi.onDriverStart(VeloxListenerApi.scala:92)
  at org.apache.gluten.backendsapi.SubstraitBackend.onDriverStart(SubstraitBackend.scala:27)
  at org.apache.gluten.backendsapi.SubstraitBackend.onDriverStart$(SubstraitBackend.scala:26)
  at org.apache.gluten.backendsapi.velox.VeloxBackend.onDriverStart(VeloxBackend.scala:52)
  at org.apache.gluten.GlutenDriverPlugin.init(GlutenPlugin.scala:65)
  at org.apache.spark.internal.plugin.DriverPluginContainer.$anonfun$driverPlugins$1(PluginContainer.scala:53)
  at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
  at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
  at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
  ...
```

Comment (majetideepak) [reply cpp/velox/udf/UdfLoader.cc:104]: It does! This check is not needed.

**Comment by majetideepak** [comment]
@zhztheplayer how do I fix the clang-format error? I don't understand the format issue.
```
cpp/velox/udf/UdfLoader.cc:80:35: error: code should be clang-formatted [-Wclang-format-violations]
      if (udfEntries == nullptr) {
                                  ^
cpp/velox/udf/UdfLoader.cc:107:36: error: code should be clang-formatted [-Wclang-format-violations]
      if (udafEntries == nullptr) {
                                   ^
Failed on file: cpp/velox/udf/UdfLoader.cc
```

**Comment by zhztheplayer** [comment]
> @zhztheplayer how do I fix the clang-format error? I don't understand the format issue.
> 
> ```
> cpp/velox/udf/UdfLoader.cc:80:35: error: code should be clang-formatted [-Wclang-format-violations]
>       if (udfEntries == nullptr) {
>                                   ^
> cpp/velox/udf/UdfLoader.cc:107:36: error: code should be clang-formatted [-Wclang-format-violations]
>       if (udafEntries == nullptr) {
>                                    ^
> Failed on file: cpp/velox/udf/UdfLoader.cc
> ```

@majetideepak You can try reformat cpp code using [this script](https://github.com/apache/incubator-gluten/blob/main/dev/formatcppcode.sh), or by running similar `clang-format` commands if the script doesn't work in your env.

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Pull request merged**